### PR TITLE
chore(table): borderless and striped, adjust styles, promote to beta

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Added borderless variant
- [x] Added striped prop
- [x] Style adjustments
- [x] Added additional stories
- [x] Added additional tests

### CODE REVIEW

File: packages/paste-core/components/table/__tests__/index.spec.tsx [hunk 1]
```diff
@@ -7,6 +7,54 @@ import {Table, THead, TBody, TFoot, Td, Tr, Th} from '../src';
...
+          <Tr>
+            <Td>Column 3</Td>
+            <Td>Column 4</Td>
+          </Tr>
+        </TBody>
+      </Table>
+    );
+    const renderedTable = screen.getByTestId('mockStripedTable').querySelector('tr:nth-of-type(even)');
+    expect(renderedTable).toHaveStyleRule('background-color', 'colorBackgroundRowStriped', {
+      target: ':nth-of-type(even)',
```
Comment (TheSisb) [inline packages/paste-core/components/table/__tests__/index.spec.tsx:55]: neat

File: packages/paste-core/components/table/src/Tr.tsx [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import * as React from 'react';
 import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {TableContext} from './TableContext';
 import {TrProps, TrPropTypes} from './types';
 
 const Tr = React.forwardRef<HTMLTableRowElement, TrProps>(({verticalAlign = 'middle', ...props}, ref) => {
+  const table = React.useContext(TableContext);
```
Comment (TheSisb) [inline packages/paste-core/components/table/src/Tr.tsx:7]: nit: not a fan of the name `table` for this const. Maybe just `{striped}`?

File: packages/paste-core/components/table/src/types.ts [hunk 3]
```diff
@@ -43,29 +53,29 @@ export const TrPropTypes = {
...
-  children: PropTypes.node.isRequired,
+  children: PropTypes.node,
   textAlign: PropTypes.oneOf(Object.values(TableAlignmentObject)),
   width: isWidthTokenProp,
 };
 
-export interface TdProps {
-  children: NonNullable<React.ReactNode>;
+export interface TdProps extends React.TdHTMLAttributes<HTMLTableCellElement> {
+  children?: React.ReactNode;
```
Comment (TheSisb) [inline packages/paste-core/components/table/src/types.ts:69]: ReactNode already handles undefined, so you can omit the `?`

File: packages/paste-core/components/table/src/types.ts [hunk 4]
```diff
@@ -43,29 +53,29 @@ export const TrPropTypes = {
   verticalAlign: PropTypes.oneOf(Object.values(TableVerticalAlignmentObject)),
 };
 
-export interface ThProps {
-  children: NonNullable<React.ReactNode>;
+export interface ThProps extends React.ThHTMLAttributes<HTMLTableHeaderCellElement> {
+  children?: React.ReactNode;
```
Comment (TheSisb) [inline packages/paste-core/components/table/src/types.ts:57]: `?` this one too

**Review by TheSisb — APPROVED** [review]
Some minor nits; this looks solid

File: packages/paste-core/components/table/src/types.ts [hunk 5]
```diff
@@ -1,39 +1,49 @@
...
 export type TableAlignmentOptions = ValueOf<typeof TableAlignmentObject>;
 export type TableLayoutOptions = ValueOf<typeof TableLayoutObject>;
+export type TableVariantOptions = ValueOf<typeof TableVariantObject>;
 export type TableVerticalAlignmentOptions = ValueOf<typeof TableVerticalAlignmentObject>;
 
+export interface TableContextProps {
+  striped: boolean;
+}
+
 export interface TableProps {
```
Comment (SiTaggart) [inline packages/paste-core/components/table/src/types.ts:10]: Not extending from the table HTML element here?

File: packages/paste-core/components/table/src/types.ts (hunk 3 above)
Comment (richbachman) [reply packages/paste-core/components/table/src/types.ts:69]: Interesting. If I remove the `?`, TS still thinks it's required.
https://share.getcloudapp.com/Wnuln1xA

Comment (richbachman) [reply packages/paste-core/components/table/src/types.ts:69]: I could just remove `children` from the interface.

File: packages/paste-core/components/table/src/types.ts (hunk 5 above)
Comment (richbachman) [reply packages/paste-core/components/table/src/types.ts:10]: Thanks for catching that.

File: packages/paste-core/components/table/src/types.ts (hunk 3 above)
Comment (richbachman) [reply packages/paste-core/components/table/src/types.ts:69]: Leaving as is for now.

**Pull request merged**